subprojects/version-control/src/main/java/org/gradle/vcs/git/internal/DefaultGitVersionControlSpec.java

```diff
@@ -52,7 +51,7 @@ public String getDisplayName() {
 
     @Override
     public String getUniqueId() {
-        return Hashing.md5().hashString(getDisplayName()).toString();
+        return getUrl().toASCIIString();
```

Comment:
I thought we wanted the implementation to be unique across version control systems? What if you could have both a GitVersionControlSystem and SimpleVersionControlSystem based on some local directory? We'd have a clash, no?

Reply:
Yep, I just moved that into VcsDependencyResolver:
https://github.com/gradle/gradle/pull/2978/files#diff-269b19ea85db319f802982807c76fb4aR117

So we take the hash of the type of system + the unique id for the spec.